src/ngraph_encapsulate_op.cc

```diff
@@ -478,13 +470,14 @@ class NGraphEncapsulateOp : public OpKernel {
             ctx, false,
             errors::Internal("Error in executing the nGraph computation\n"));
       }
+      BackendManager::UnlockBackend(m_op_backend_name);
```

Comment:
Slightly unsafe that we are locking/unlocking by hand. But there is a catch-all exception here, so no error should happen which would leave the mutex in a locked state.

Reply:
Yes, and the next level of refactoring/design will get rid of this, and move most of this to Backend class